USAGE.md

```diff
@@ -0,0 +1,274 @@
+
+## Setup and Install
+
+### Prerequisites
+
+* [Python 3.6+](https://www.python.org/)
+* [make](https://www.gnu.org/software/make/)
+* [pip](https://pypi.org/project/pip/)
+* [git](https://git-scm.com/)
+
+#### Clone from GitHub
+
+The recommended way to download the ECS repo is `git clone`:
+
+```
+$ git clone https://github.com/elastic/ecs
+$ cd ecs
+```
+
+Prior to installing dependencies or running the tools, it's recommended to check out the `git` branch for the ECS version being targeted.
+
+**Example**: For ECS `1.5.0`:
+
+```
+$ git checkout v1.5.0
+```
+
+#### Option 1: Install dependencies via make (recommended)
+
+Setting up a `virtualenv` (`venv`) can be accomplished by running `make ve` the top-level of the ECS repo:
+
+```
+$ make ve
+```
+
+All necessary Python dependencies will also be installed with `pip`.
+
+#### Option 2: Install dependencies via pip
+
+Install dependencies using `pip` (An active `virutalenv` is recommended):
+
+```
+$ pip install -r scripts/requirements.txt
+```
+
+## Usage
+
+### Getting Started - Generating Artifacts
+
+Using the defaults, the [generator](scripts/generator.py) script generates the artifacts based on the [current](schemas) ECS schema.
+
+```
+$ python scripts/generator.py
+Loading schemas from local files
+Running generator. ECS version 1.5.0
+```
+
+**Points to note on the defaults**:
+
+* Artifacts are created in the [`generated`](generated) directory and the entire schema is included
+* Documentation updates will be written to the appropriate file under the `docs` directory. More specifics on generated doc files is covered in the [contributor's file](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md#generated-documentation-files)
+* Each run of the script will rewrite the entirety of the `generated` directory
+* The script will need to be executed from the top-level of the ECS repo
+* The `version` displayed when running `generator.py` is based on the current value of the [version](version) file in the top-level of the repo
+
+The following options add functionality beyond the defaults.
+
+### Generator Options
+
+#### Out
+
+Generate the ECS artifacts in a different output directory. The argument expects to be passed an existing directory:
+
+```
+$ python scripts/generator.py --out ../myproject/ecs/out/
+```
+
+Inside the directory passed in as the target dir to the `--out` flag, two directories, `generated` and `docs`, will be created. `docs` will contain three asciidoc files based on the contents of the provided schema. `generated` will contain the various artifacts laid out as in the published repo (`beats`, `csv`, `ecs`, `elasticsearch`).
+
+#### Include
+
+Use the `--include` flag to generate ECS artifacts based on the current ECS schema field definitions plus provided custom fields:
+
+```
+$ python scripts/generator.py --out ../myproject/ecs/custom-fields/
```

Comment:
I think this might have been a copy/paste issue `--out` should be `--include` here.

Reply:
Fixed.